instrumentation/net/http/otelhttp/test/handler_test.go

```diff
@@ -468,3 +468,54 @@ func TestSpanStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestWithRouteTag(t *testing.T) {
+	route := "/some/route"
+
+	spanRecorder := tracetest.NewSpanRecorder()
+	tracerProvider := sdktrace.NewTracerProvider()
+	tracerProvider.RegisterSpanProcessor(spanRecorder)
+
+	metricReader := metric.NewManualReader()
+	meterProvider := metric.NewMeterProvider(metric.WithReader(metricReader))
+
+	h := otelhttp.NewHandler(
+		otelhttp.WithRouteTag(
+			route,
+			http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusTeapot)
+			}),
+		),
+		"test_handler",
+		otelhttp.WithTracerProvider(tracerProvider),
+		otelhttp.WithMeterProvider(meterProvider),
+	)
+
+	h.ServeHTTP(httptest.NewRecorder(), httptest.NewRequest(http.MethodGet, "/", nil))
+	expectedAttribute := semconv.HTTPRouteKey.String(route)
+
+	require.Len(t, spanRecorder.Ended(), 1, "should emit a span")
+	span := spanRecorder.Ended()[0]
+	require.Contains(t, span.Attributes(), expectedAttribute, "should add route to span attributes")
+
+	rm := metricdata.ResourceMetrics{}
+	err := metricReader.Collect(context.Background(), &rm)
+	require.NoError(t, err)
+	require.Len(t, rm.ScopeMetrics, 1, "should emit metrics for one scope")
+	scopeMetrics := rm.ScopeMetrics[0]
+
+	for _, m := range scopeMetrics.Metrics {
+		switch d := m.Data.(type) {
+		case metricdata.Sum[int64]:
+			require.Len(t, d.DataPoints, 1, "metric '%v' should have exactly one data point", m.Name)
+			require.Contains(t, d.DataPoints[0].Attributes.ToSlice(), expectedAttribute, "should add route to attributes for metric '%v'", m.Name)
+
+		case metricdata.Histogram[float64]:
+			require.Len(t, d.DataPoints, 1, "metric '%v' should have exactly one data point", m.Name)
+			require.Contains(t, d.DataPoints[0].Attributes.ToSlice(), expectedAttribute, "should add route to attributes for metric '%v'", m.Name)
+
+		default:
+			require.Fail(t, "metric has unexpected data type", "metric '%v' has unexpected data type %T", m.Name, m.Data)
```

Comment:
Do I understand correctly that this test would start failing if `otelhttp` will produce new metrics (of types other than `metricdata.Histogram[float64]` and `metricdata.Sum[int64]`)?

If I am correct then I think that this test would be too strict and I would propose to remove this `default` case.

Reply:
This is intentional - the expectation is that the route tag is applied to all metrics produced by `otelhttp`, so if a new metric is added that isn't an expected type, this test should fail and prompt the person adding the new type of metric to expand this test to cover that kind of metric.

Reply:
>  so if a new metric is added that isn't an expected type, this test should fail and prompt the person adding the new type of metric to expand this test to cover that kind of metric.

I disagree that `TestWithRouteTag` should fail when one adds a new type of metric to be produced by `otelhttp`.
I find it as an "[overspecified](http://xunitpatterns.com/Fragile%20Test.html#Overspecified%20Software) test". This test should not test what types of metrics are produced by `otelhttp`. 

As an alternative, you could update the test to assert against all possible metrics types. If you extract such logic to a helper assertion function it could be used in other tests as well.

Reply:
>As an alternative, you could update the test to assert against all possible metrics types. 

I've added this in fc1ec77.